pkg/client/client.go

```diff
@@ -363,7 +363,31 @@ func (c *client) List(ctx context.Context, obj ObjectList, opts ...ListOption) e
 
 	switch x := obj.(type) {
 	case runtime.Unstructured:
-		return c.unstructuredClient.List(ctx, obj, opts...)
+		// Call the list client.
+		if err := c.unstructuredClient.List(ctx, obj, opts...); err != nil {
+			return err
+		}
+
+		gvk := obj.GetObjectKind().GroupVersionKind()
+		if !c.scheme.Recognizes(gvk) {
+			// If the scheme doesn't recognize the GVK, it will not be set on
+			// the list items. We can however produce an unsafe guess which
+			// will work for most cases, and eases consumption.
+			itemGVK := schema.GroupVersionKind{
```

Comment:
Are you sure that this is set in the structured case? I thought this is never the case due to https://github.com/kubernetes/kubernetes/issues/80609